### PR TITLE
Update API docs for new get_finalizers_info_rpc

### PR DESCRIPTION
**Pull request by ericpassmore**
Adds description for return type of new `get_finalizers_info` endpoint in `chain_api` plugin

### CODE REVIEW

File: plugins/chain_api_plugin/chain.swagger.yaml [hunk 1]
```diff
@@ -89,6 +89,59 @@ paths:
...
+      operationId: get_finalizer_info_result
+        responses:
+          "200":
+            description: OK
+            content:
+              application/json:
+                schema:
+                  active_finalizer_policy:
+                    type: array
+                    description: Variant type, an array of strings with the current finalizers
```
Comment (heifner) [inline plugins/chain_api_plugin/chain.swagger.yaml:104]: I think I would remove `Variant type, ` that is an implementation detail. The JSON is an object with `generation`, `threshold`, and an array of `finalizers`. Seems like that all should be described.
```
 "active_finalizer_policy": {
      "generation": 1,
      "threshold": 3,
      "finalizers": [
        {
          "description": "finalizer #1",
          "weight": 1,
          "public_key": "PUB_BLS_PcJG7lYgwZcsFGe-i4WWYDw1GtP7r5vGWFcRe97TVcp5f5_h6RRRPRhZSGKtRF4CtdJxAP50VewH4VIhMIPfhsu9C4GGH4MgOzSwl51N3EK6XiAiRHQ_lcGvxggipYwIReHkhQ"
        },
```

**Review by heifner — CHANGES_REQUESTED** [review]

File: plugins/chain_api_plugin/chain.swagger.yaml [hunk 2]
```diff
@@ -100,15 +100,31 @@ paths:
...
+                        type: array
+                        items:
+                          ref: "#/components/schemas/Finalizer"
                   pending_finalizer_policy:
-                    type: array
-                    description: Variant type, an array of strings with the pending finalizers
-                    items:
-                      type: string
+                    type: object
+                    description: List of current finalizers
```
Comment (heifner) [inline plugins/chain_api_plugin/chain.swagger.yaml:117]: List of pending finalizers

File: plugins/chain_api_plugin/chain.swagger.yaml [hunk 3]
```diff
@@ -89,6 +89,75 @@ paths:
...
+                      threshold:
+                        type: integer
+                        description: total weight needed to advance finality
+                      finalizers:
+                        type: array
+                        items:
+                          ref: "#/components/schemas/Finalizer"
+                  last_tracked_votes:
+                    type: array
+                    description: Record of last votes for top 21 finalizers
```
Comment (linh2931) [inline plugins/chain_api_plugin/chain.swagger.yaml:130]: Not top 21 finalizers. We return the last votes for the finalizers in current active finalizer policy and pending finalizer policy.

**Comment by ericpassmore** [comment]
Moving Out of Draft 

Updated on Staging 
https://staging.docs.eosnetwork.com/apis/spring/latest/chain.api/#operation/get_block_info

Screenshot 2024-08-01 at 1 06 24 PM

**Review by heifner — APPROVED** [review]

File: plugins/chain_api_plugin/chain.swagger.yaml [hunk 4]
```diff
@@ -89,6 +89,171 @@ paths:
             application/json:
               schema:
                 $ref: "https://eosio.github.io/schemata/v2.1/oas/BlockInfo.yaml"
+  /get_finalizer_info_result:
```
Comment (linh2931) [inline plugins/chain_api_plugin/chain.swagger.yaml:92]: should be `/get_finalizer_info:`

File: plugins/chain_api_plugin/chain.swagger.yaml [hunk 5]
```diff
@@ -89,6 +89,171 @@ paths:
...
+    post:
+      description: Returns information about finalizers
+      operationId: get_finalizer_info_result
```
Comment (linh2931) [inline plugins/chain_api_plugin/chain.swagger.yaml:95]: `get_finalizer_info`

File: plugins/chain_api_plugin/chain.swagger.yaml [hunk 6]
```diff
@@ -89,6 +89,171 @@ paths:
...
+                        example: 15
+                      finalizers:
+                        type: array
+                        items:
+                          type: object
+                          description: Description of a finalizer
+                          properties:
+                            'description':
+                              type: string
+                              description: Name of finalizer, typicaly a producer acount
```
Comment (linh2931) [inline plugins/chain_api_plugin/chain.swagger.yaml:131]: `typically` not `typicaly`, `account` not `acount`?

File: plugins/chain_api_plugin/chain.swagger.yaml [hunk 7]
```diff
@@ -89,6 +89,171 @@ paths:
...
```
Comment (linh2931) [inline plugins/chain_api_plugin/chain.swagger.yaml:161]: `typically`, `acount`

File: plugins/chain_api_plugin/chain.swagger.yaml [hunk 8]
```diff
@@ -89,6 +89,171 @@ paths:
...
+                          type: boolean
+                          description: When true indicate strongly voted
+                        voted_policy_generation:
+                          type: integer
+                          description: Policy Generation for vote
+                          example: 2
+                        voted_block_id:
+                          type: string
+                          description: Block id voted on
+                          example: 1234
```
Comment (linh2931) [inline plugins/chain_api_plugin/chain.swagger.yaml:195]: Examples of `voted_block_id` and `voted_block_num` are reverted.

**Review by linh2931 — CHANGES_REQUESTED** [review]

File: plugins/chain_api_plugin/chain.swagger.yaml [hunk 9]
```diff
@@ -89,6 +89,171 @@ paths:
...
+              schema:
+                type: object
+                properties:
+                  active_finalizer_policy:
+                    type: object
+                    description: List of current finalizers
+                    properties:
+                      generation:
+                        type: integer
+                        description: version number of finalizers set
```
Comment (greg7mdp) [inline plugins/chain_api_plugin/chain.swagger.yaml:117]: generation number of finalizer policy. Activated finalizer policy generation numbers are always increasing, but may have gaps.

**Review by greg7mdp — APPROVED** [review]

**Review by linh2931 — COMMENTED** [review]
The several field names have been changed by the committed version of https://github.com/AntelopeIO/spring/pull/453:

`voted_policy_generation` -> finalizer_policy_generation`
`voted_block_id` -> `voted_for_block_id`
`voted_block_num` -> `voted_for_block_num`
`voted_block_timestamp` -> `voted_for_block_timestamp`

Please update the document accordingly.

**Comment by ericpassmore** [comment]
@linh2931 Updated PR and updated on Staging Docs Server.

**Comment by ericpassmore** [comment]
Screenshot 2024-08-02 at 2 23 31 PM

File: plugins/chain_api_plugin/chain.swagger.yaml [hunk 10]
```diff
@@ -89,6 +89,171 @@ paths:
...
+                              type: string
+                              description: The BLS Public Key used by this finalizer
+                              example: "PUB_BLS_PcJG7lYgwZcsFGe-i4WWYDw1GtP7r5vGWFcRe97TVcp5f5_h6RRRPRhZSGKtRF4CtdJxAP50VewH4VIhMIPfhsu9C4GGH4MgOzSwl51N3EK6XiAiRHQ_lcGvxggipYwIReHkhQ"
+                  last_tracked_votes:
+                    type: array
+                    description: Record of last votes for the finalizers in current active finalizer policy and pending finalizer policy.
+                    items:
+                      type: object
+                      properties:
+                        public_key:
```
Comment (linh2931) [inline plugins/chain_api_plugin/chain.swagger.yaml:177]: `description` is now the first field, `public_key` is the second. The order of the rest is correct.

**Review by linh2931 — APPROVED** [review]

**Comment by ericpassmore** [comment]
Note:start
group: RELEASE
category: CHORE
summary: API docs on get_finalizer_info_rpc.
Note:end

**Pull request merged**